Services/Simulation/JavascriptInterpreter.cs

```diff
@@ -178,26 +192,34 @@ private void UpdateState(JsValue data)
         {
             string key;
             object value;
-            Dictionary<string, object> stateChanges;
+            Dictionary<string, object> stateChanges = this.JsValueToDictionary(data);
 
             this.log.Debug("Updating state from the script", () => new { data, this.deviceState });
 
-            stateChanges = this.JsValueToDictionary((JsValue) data);
-
-            //Update device state with the script data passed
+            // Update device state with the script data passed
             lock (this.deviceState)
```

Comment:
do you need the lock anymore?  You're using a concurrent dictionary for this now.